redhat-appstudio/service-provider-integration-oauth
# Token upload

### What does this PR do?
This implements the endpoint for manual token upload. Depends on https://github.com/redhat-appstudio/service-provider-integration-oauth/pull/62

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-62


### How to test this PR?
1. in operator repo:
1. `make install deploy_minikube SPIS_IMG=quay.io/lkrejci/service-provider-integration-oauth:svpi-62 SPIO_IMG=quay.io/lkrejci/service-provider-integration-operator:svpi-62`
1. `hack/edit-spi-config.sh` and fill in the clientId and clientSecret for your Github OAuth app, specify the shared secret.
1. `kubectl -n spi-system scale deployment spi-controller-manager spi-oauth-service --replicas=0`
1. `kubectl -n spi-system scale deployment spi-controller-manager spi-oauth-service --replicas=1`
1. `kubectl apply -f hack/give-default-sa-perms-for-accesstokens.yaml`
1. `kubectl apply -f samples/binding-write-repo.yaml`
1. `TOKEN_NAME=$(kubectl get spiaccesstokenbinding binding-write-repository -o jsonpath='{.status.linkedAccessTokenName}'`
1. `curl -v -k -H "Authorization: Bearer $(./hack/get-default-sa-token.sh)" "http://spi.$(minikube ip).nip.io/token/default/$TOKEN_NAME" -d '{"access_token": "42"}'`
1. `kubectl get spiaccesstoken $TOKEN_NAME -o yaml`
1. Check that the token is in `Invalid` phase with the error message detailing the error

This proves that the token data was uploaded.
